client/extensions/woocommerce/app/settings/email/mailchimp/setup-mailchimp.js

```diff
@@ -0,0 +1,315 @@
+	'store_currency_code', 'admin_email' ];
+
+const campaignDefaultsRequiredFields = [ 'campaign_from_name', 'campaign_from_email', 'campaign_subject',
+		'campaign_language', 'campaign_permission_reminder' ];
+
+class MailChimpSetup extends React.Component {
+
+	constructor( props ) {
+		super( props );
+		// make this react to the real phase the execution is.
+		this.state = {
+			step: LOG_INTO_MAILCHIMP_STEP,
+			settings: this.prepareDefaultValues( this.props.settings ),
+			settings_values_missing: false,
+			api_key_input: this.props.settings.mailchimp_api_key || '',
+		};
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( nextProps.settings.mailchimp_lists && ! this.state.settings.mailchimp_lists ) {
+			const newSettings = Object.assign( {}, this.state.settings );
+			newSettings.mailchimp_lists = nextProps.settings.mailchimp_lists;
+			newSettings.mailchimp_list = nextProps.settings.mailchimp_list;
+			this.setState( { settings: newSettings } );
+		}
+
+		// No state changes while doing request.
+		if ( nextProps.isBusy ) {
+			return;
+		}
+
+		if ( ( nextProps.settings.active_tab === STORE_INFO_STEP ) &&
+			( this.state.step === KEY_INPUT_STEP ) ) {
+			this.setState( { step: STORE_INFO_STEP } );
+			const settings = this.prepareDefaultValues( nextProps.settings );
+			settings.admin_email = nextProps.settings.admin_email || nextProps.currentUserEmail || '';
+			settings.store_timezone = nextProps.settings.store_timezone || nextProps.timezone || 'America/New_York';
+			this.setState( { settings } );
+		} else if ( ( nextProps.settings.active_tab === CAMPAIGN_DEFAULTS_STEP ) &&
+			( this.state.step === STORE_INFO_STEP ) ) {
+			this.setState( { step: CAMPAIGN_DEFAULTS_STEP } );
+			this.setState( { settings: this.prepareDefaultValues( nextProps.settings ) } );
+		} else if ( ( nextProps.settings.active_tab === NEWSLETTER_SETTINGS_STEP ) &&
+			( this.state.step === CAMPAIGN_DEFAULTS_STEP ) ) {
+			this.setState( { step: NEWSLETTER_SETTINGS_STEP } );
+		} else if ( ( nextProps.settings.active_tab === STORE_SYNC ) &&
+			( this.state.step === NEWSLETTER_SETTINGS_STEP ) ) {
+			nextProps.onClose();
+		}
+	}
+
+	prepareDefaultValues( settings ) {
+		const newSettings = Object.assign( {}, settings );
+		newSettings.campaign_from_name = settings.campaign_from_name || settings.store_name || '';
+		newSettings.campaign_from_email = settings.campaign_from_email || settings.admin_email || '';
+		newSettings.campaign_subject = settings.campaign_subject || settings.store_name || '';
+		newSettings.store_locale = settings.store_locale || 'en';
+		newSettings.campaign_language = settings.campaign_language || settings.store_locale;
+		newSettings.campaign_permission_reminder = settings.campaign_permission_reminder ||
+			'You were subscribed to the newsletter from ' + settings.store_name;
+		return newSettings;
+	}
+
+	onClose = () => {
+		this.props.onClose();
```

Comment:
Another good spot for a default prop of `noop` or check for the method before calling it.

Reply:
let's just make it required.